shelley/chain-and-ledger/executable-spec/test/Generator/LedgerTrace.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+-- The HasTrace instance relies on test generators and so cannot
```

Comment:
I don't get this. Would it be wrong to add the `HasTrace` instance to the same module where the `LEDGER` STS is defined? Maybe the extra dependency on `hedgehog` :thinking:

Reply:
I understand the reason and yes, having a hedgehog dependency for the STS rules themselves isn't great. But then is having orphan instances?

Reply:
👍  deferred to #891